src/Azure.Functions.Cli/Actions/AzureActions/CreateAADApplication.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Common;
+using Azure.Functions.Cli.Interfaces;
+using Fclp;
+namespace Azure.Functions.Cli.Actions.AzureActions
+{
+    // Invoke via `func azure auth create-aad -aad-name {displayNameOfAAD} --app-name {displayNameOfApp}`
+    [Action(Name = "create-aad", Context = Context.Azure, SubContext = Context.Auth, HelpText = "Creates a production Azure Active Directory application with given name. Can be linked to specified Azure Application")]
+    class CreateAADApplication : BaseAzureAction
+    {
+        private readonly IAuthManager _authManager;
+
+        public string AADName { get; set; }
```

Comment:
Naming nit: "AAD" is not really a thing, so `AADName` sounds unnatural. I'm also not sure what the difference is between an "AADName" and an "AppName". Should they be the same? Or should "AADName" be renamed to "AppRegistrationName"?

Reply:
Same comment for the parameter syntax below.

Reply:
What are your thoughts on AADRegistrationName for both the variable and what goes on the command line? And for Azure App Service apps/functions, appName?

Reply:
Maybe `AADAppRegistrationName` instead of AADRegistrationName? `appName` is fine for the function app name.